packages/main/src/Breadcrumbs.ts

```diff
@@ -497,16 +578,43 @@ class Breadcrumbs extends UI5Element {
 	 */
 	get _linksData() {
 		const items = this._visibleItems;
-		const itemsCount = items.length;
+		const itemsCount = items.length; // get size before removing of current location
+
+		if (this._endsWithCurrentLocation && !this._endsWithCurrentLinkItem) {
+			items.pop();
+		}
 
 		return items
 			.map((item, index) => {
 				item._accessibleNameText = this._getItemAccessibleName(item, index + 1, itemsCount);
-				item._isCurrentPageItem = index === (itemsCount - 1) && this._endsWithCurrentPageItem;
+				item._isCurrentPageItem = index === (itemsCount - 1) && !!this._endsWithCurrentLocation;
+				item._needsSeparator = !item._isCurrentPageItem || (!!this._endsWithCurrentLinkItem && !this._endsWithCurrentLocation);
```

Comment:
the !! that precedes  this._endsWithCurrentLocation seems redundant as the function already returns boolean